### ISSUE TITLE
If the server is started via Systemd, then it will not start when rebooting via RCON.

### ISSUE THREAD

**Issue by Timofei302** [TeamHypersomnia/Hypersomnia#279]
Systemd service config:
![Screenshot_20231017_190858](https://github.com/TeamHypersomnia/Hypersomnia/assets/59145732/8c2b4e54-207c-42e8-8a58-1902fe875b83)

**Comment by geneotech** [comment]
I see. I haven't really designed it yet to be run as a service, will have to look into it later. Would probably need a separate flag ``--as-service`` and have a separate restart logic for it. Do you think a service can update and restart itself though?

**Comment by geneotech** [comment]
Actually wouldn't it be enough for the game to ``return 1;`` since you've set ``Restart=on-failure``?

**Comment by Timofei302** [comment]
> Actually wouldn't it be enough for the game to `return 1;` since you've set `Restart=on-failure`?

Logically, this should be enough, but I think it's worth doing research first by making a script that will return exit code 1 and see how systemd behaves.

**Comment by geneotech** [comment]
Hey, I think your server keeps restarting or perhaps has a high ``send_heartbeat_to_server_list_once_every_secs`` set (I should probably clamp the value automatically), since the server list detects it as a new server once every few minutes, and it keeps spamming new server notifications :) The timeout for a server to be removed from the server list is 60 seconds.

**Comment by Timofei302** [comment]
> Hey, I think your server keeps restarting or perhaps has a high ``send_heartbeat_to_server_list_once_every_secs`` set (I should probably clamp the value automatically), because the server list detects it as a new server once every few minutes, and it keeps spamming new server notifications :) The timeout for a server to be removed from the server list is 60 seconds.

idk what caused this, since everything is fine with my settings (at least everything was fine), and I am not reconfiguring the server now (so that when I configure the server, I often reboot hypersomnia via systemctl).

![Screenshot_20231017-212317.jpg](https://github.com/TeamHypersomnia/Hypersomnia/assets/59145732/159397fd-a071-402a-b872-a5cdf66995b9)

**Comment by Timofei302** [comment]
![Screenshot_20231017-213115.jpg](https://github.com/TeamHypersomnia/Hypersomnia/assets/59145732/1eadcafa-9bb7-4c15-a610-11ca92715b4e)

**Comment by geneotech** [comment]
It seems it now sends the heartbeats correctly :+1:

**Comment by Timofei302** [comment]
The server still goes missing every 5 minutes :/

**Comment by geneotech** [comment]
Yeah, just noticed, not sure what could be the cause, perhaps packet loss? What if you set ``send_heartbeat_to_server_list_once_every_secs`` to e.g. 3?

**Comment by Timofei302** [comment]
> Yeah, just noticed, not sure what could be the cause, perhaps packet loss? What if you set ``send_heartbeat_to_server_list_once_every_secs`` to e.g. 3?

Hmm. By default, my server is already configured to send a "heartbeat" every 10 seconds. 
![Screenshot_20231017-220755.jpg](https://github.com/TeamHypersomnia/Hypersomnia/assets/59145732/bc15ece4-00ce-4334-b650-ca8046e67682)


Therefore, there are two possible reasons - my hosting provider is still under ddos attack and this has started to negatively affect the operation of my vds, or someone or something is trying to break the connection of my vds with your master server on your side.


I can't think of any other assumptions.
